### PR TITLE
Fix broken links in README.md

### LINKED ISSUE #334: Fix broken links in readme algorithm list
Many of the links to specific algorithms in the [readme](https://github.com/ProAlgos/ProAlgos-Cpp/tree/master/C%2B%2B) are broken. It appears to mostly be sorting algorithms.

- [ ] Check that links direct to appropriate algorithm implementation (which will be a header file ending in `.hpp`).
- [ ] Fix broken links.

### PR DESCRIPTION
**Pull request by a-moreira**
Fixed broken links in README.md

- [x] Read the [contribution guidelines][contrib-guidelines]
- [x] Added [unit tests][unit-tests] (or unit tests have already been added)
- [x] Updated the [Contents][contents]

Closes #334

[contrib-guidelines]: https://github.com/ProAlgos/ProAlgos-Cpp/blob/master/.github/CONTRIBUTING.md
[unit-tests]: https://github.com/ProAlgos/ProAlgos-Cpp/blob/master/C++/CONTRIBUTING.md#unit-tests
[contents]: https://github.com/ProAlgos/ProAlgos-Cpp/tree/master/C%2B%2B#contents

### CODE REVIEW

File: C++/README.md [hunk 1]
```diff
@@ -41,24 +41,25 @@ Implementing well-known (and some rare) **algorithms and data structures in C++*
...
+  * [Selection sort](include/algorithm/sorting/selection_sort.hpp) :white_check_mark:
+  * [Shell sort](include/algorithm/sorting/shell_sort.hpp) :white_check_mark:
 
 * String
   * [Longest common subsequence](include/algorithm/string/longest_common_subsequence.hpp) :white_check_mark:
   * Searching (pattern matching)
     * [Knuth-Morris-Pratt](include/algorithm/string/knuth_morris_pratt.hpp) :white_check_mark:
   * [Edit Distance Problem](include/algorithm/string/edit_distance.hpp) :white_check_mark:
   * [Shunting yard](include/algorithm/string/shunting_yard.hpp) :white_check_mark:
+  * [Heap's Algorithm](include/algorithm/string/heaps_algorithm.hpp) :white_check_mark:
```
Comment (alxmjo) [inline C++/README.md]: Move this change into #337.

**Review by alxmjo — CHANGES_REQUESTED** [review]
This looks great. Thank you. See my only comment about migrating the listing for Heap's.

**Comment by alxmjo** [comment]
This branch now has merge conflicts since Heap's was also added to readme in #337. Please resolve.

**Comment by a-moreira** [comment]
I think everything is set up now

**Review by alxmjo — APPROVED** [review]

**Comment by alxmjo** [comment]
Thanks!

**Pull request merged**